src/explore-education-statistics-admin/src/pages/release/data/components/ApiDataSetAutoMappedLocationsTable.tsx

```diff
@@ -1,61 +1,206 @@
-import { AutoMappedLocation } from '@admin/pages/release/data/utils/getApiDataSetLocationMappings';
+import {
+  AutoMappedLocation,
+  LocationCandidateWithKey,
+} from '@admin/pages/release/data/utils/getApiDataSetLocationMappings';
 import styles from '@admin/pages/release/data/ReleaseApiDataSetLocationsMappingPage.module.scss';
-import ButtonText from '@common/components/ButtonText';
+import { PendingLocationMappingUpdate } from '@admin/pages/release/data/ReleaseApiDataSetLocationsMappingPage';
+import ApiDataSetLocationMappingModal from '@admin/pages/release/data/components/ApiDataSetLocationMappingModal';
+import getApiDataSetLocationCodes from '@admin/pages/release/data/utils/getApiDataSetLocationCodes';
 import Tag from '@common/components/Tag';
+import ButtonText from '@common/components/ButtonText';
+import Pagination from '@common/components/Pagination';
+import LoadingSpinner from '@common/components/LoadingSpinner';
+import FormSearchBar from '@common/components/form/FormSearchBar';
 import VisuallyHidden from '@common/components/VisuallyHidden';
-import React from 'react';
+import useDebouncedCallback from '@common/hooks/useDebouncedCallback';
+import { LocationLevelKey } from '@common/utils/locationLevelsMap';
+import chunk from 'lodash/chunk';
+import React, { useMemo, useState } from 'react';
+
+const locationsPerPage = 10;
+const formId = 'auto-mapped-search';
 
 interface Props {
+  level: LocationLevelKey;
   locations: AutoMappedLocation[];
+  newLocations: LocationCandidateWithKey[];
+  pendingUpdates?: PendingLocationMappingUpdate[];
+  onUpdate: (update: PendingLocationMappingUpdate) => Promise<void>;
 }
 
 export default function ApiDataSetAutoMappedLocationsTable({
+  level,
   locations,
+  newLocations,
+  pendingUpdates = [],
+  onUpdate,
 }: Props) {
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [searchTerm, setSearchTerm] = useState<string>();
+
+  const filteredLocations = useMemo(() => {
+    return searchTerm
+      ? locations.filter(location => {
+          const { candidate, mapping } = location;
+          const searchFields = [
+            candidate.label.toLowerCase(),
+            candidate.code?.toLowerCase(),
+            candidate.laEstab?.toLowerCase(),
+            candidate.oldCode?.toLowerCase(),
+            candidate.ukprn?.toLowerCase(),
+            candidate.urn?.toLowerCase(),
+            mapping.source.label.toLowerCase(),
+            mapping.source.code?.toLowerCase(),
+            mapping.source.laEstab?.toLowerCase(),
+            mapping.source.oldCode?.toLowerCase(),
+            mapping.source.ukprn?.toLowerCase(),
+            mapping.source.urn?.toLowerCase(),
+          ];
```

Comment:
This will create a lot of additional array entries that have to be searched on every location reducing search performance. 

To avoid this, we should filter out any empty strings or undefined values e.g.
```ts
import compact from 'lodash/compact';

const searchFields = compact([
  ...
]);
```